### PR TITLE
Add ZoneMinder and Gotify basic collections for bruteforce

### PR DESCRIPTION
**Pull request by baudneo**
- ZoneMinder collection with parser and simple bruteforce scenario.
- Gotify collection with parser and simple bruteforce scenario.

### CODE REVIEW

File: .tests/zoneminder-bf/results/bucket-dump.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (blotus) [inline .tests/zoneminder-bf/results/bucket-dump.yaml:1]: Please remove this file

File: .tests/zoneminder-bf/results/bucketpour-dump.yaml [hunk 2]
```diff
@@ -0,0 +1,394 @@
+OK:
```
Comment (blotus) [inline .tests/zoneminder-bf/results/bucketpour-dump.yaml:1]: Please remove this file

File: .tests/zoneminder-web_php/results/bucket-dump.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (blotus) [inline .tests/zoneminder-web_php/results/bucket-dump.yaml:1]: Please remove this file

File: .tests/zoneminder-web_php/results/bucketpour-dump.yaml [hunk 4]
```diff
@@ -0,0 +1,394 @@
+OK:
```
Comment (blotus) [inline .tests/zoneminder-web_php/results/bucketpour-dump.yaml:1]: Please remove this file

File: .tests/zoneminder-web_php/results/parser-dump.yaml [hunk 5]
```diff
@@ -0,0 +1,659 @@
+s00-raw:
```
Comment (blotus) [inline .tests/zoneminder-web_php/results/parser-dump.yaml:1]: Please remove this file

File: .tests/zoneminder-bf/results/parser-dump.yaml [hunk 6]
```diff
@@ -0,0 +1,659 @@
+s00-raw:
```
Comment (blotus) [inline .tests/zoneminder-bf/results/parser-dump.yaml:1]: Please remove this file

File: parsers/s01-parse/baudneo/gotify-logs.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+#filter: '1 == 1'  # For hub tests
+filter: evt.Parsed.program == "gotify"  # Production
+#debug: true
+onsuccess: next_stage
+name: baudneo/gotify-logs
+description: parser for Gotify server
+pattern_syntax:
+  GOTIFY_DATE: '%{YEAR}/%{MONTHNUM}/%{NUMBER}%{SPACE}[-]%{SPACE}%{TIME}'
+    GOTIFY_SEPERATOR: '%{SPACE}\|%{SPACE}'
```
Comment (blotus) [inline parsers/s01-parse/baudneo/gotify-logs.yaml:9]: Seems the indentation is not correct ?

File: parsers/s01-parse/baudneo/gotify-logs.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    #  GOTIFY_DATE: \d*/\d*/\d*\s[\-]\s.*
+    #  GOTIFY_SEPERATOR: \s*\|\s*
+    GOTIFY_401: '^\[GIN\]\s*%{GOTIFY_DATE}%{GOTIFY_SEPERATOR}401%{GOTIFY_SEPERATOR}%{DATA:request_time_took}%{GOTIFY_SEPERATOR}%{IP:source_ip}%{GOTIFY_SEPERATOR}%{WORD:request_type}%{SPACE}"%{DATA:endpoint}"'
```
Comment (blotus) [inline parsers/s01-parse/baudneo/gotify-logs.yaml:12]: Seems the indentation is not correct ?

**Comment by blotus** [comment]
Hello @baudneo,

Thanks for the PR, I've added some comments, but it looks good to me.

File: parsers/s01-parse/baudneo/gotify-logs.yaml (hunk 8 above)
Comment (baudneo) [reply parsers/s01-parse/baudneo/gotify-logs.yaml]: I have corrected the indentation and matched the static Meta.source_ip to the GROK pattern.

**Comment by blotus** [comment]
I missed the empty assertions for the parsers and scenarios @baudneo, which make the tests fail.

Can you fill them properly ?

You also did not delete the `.tests/zoneminder-web_php/results/` directory.

**Comment by baudneo** [comment]
How do I create the assertions? I followed the directions in the docs for creating a parser, the docs said that the `cscli run <test>` would create the assertions? When I run the tests it says to create the assertions? The docs in the README of the hub say to use .test.sh but the documentation for creating a parser say to use `cscli hubtest`? I used the `cscli hubtest create` command to create the test config and files.

**Comment by baudneo** [comment]
I followed the instructions in the hub-test repo, setup the environment then created a test. The test folder is in ./parsers/s01-parse/baudneo/.tests/zoneminder-web_php. It has a config.yaml file and I added a test.log file, when I run a --single test it errors out ->

```
ERRO[0000] invalid baudneo/zoneminder-web_php of type parsers
ERRO[0000] item might be missing or misnamed
FATA[0000] provided with empty filename
```

I am assuming this is because my parser is not in the ./config/hub/index.json file. How do I add my parser there? Is there any docs that have complete instructions on how to add and configure a parser and run test?

**Comment by sabban** [comment]
Please have a look at this PR https://github.com/baudneo/hub/pull/1 IF you are ok with that and merge it, we should be able to merge it back here

Regards,

**Pull request merged**